internal/graderd/scheduler.go

```diff
@@ -2,20 +2,18 @@ package graderd
 
 import (
 	"context"
-	"math/rand"
-
 	"github.com/pkg/errors"
+	"math/rand"
+	"github.com/Capstone-auto-grader/grader-api-v2/internal/dockerdb"
+	graderTask "github.com/Capstone-auto-grader/grader-api-v2/internal/grader-task"
 )
 
 type Scheduler interface {
 	CreateImage(ctx context.Context, imageName string, imageTar []byte) error
-
-	ListTasks(ctx context.Context, assignmentID string, db Database) ([]*Task, error)
-	CreateTasks(ctx context.Context, taskList []*Task, db Database) error
-	StartTasks(ctx context.Context, taskList []*Task, db Database) error
-	EndTask(ctx context.Context, taskID string, db Database) error
-
-	TaskOutput(ctx context.Context, taskID string, db Database) ([]byte, error)
+	ListTasks(ctx context.Context) ([]*graderTask.Task, error)
+	StartTask(ctx context.Context, taskID string) error
+	EndTask(ctx context.Context, taskID string) error
+	TaskOutput(ctx context.Context, taskID string) ([]byte, error)
```

Comment:
I have been contemplating a lot with the signature of these functions lately... more specifically with using `taskID` string as an identifier. I know that we use `taskID` for all the lookups internally (`string -> *Task`) using some storage. But since our implementation of the storage is in-mem, we don't really need to constraint ourselves to string lookups, instead we can use `*Task` as key. By doing that we can be more aligned with the `Database` interface and simplify our code quite a bit here and there. Anyways, I want to hear what you think about this.